ydb/core/tx/schemeshard/schemeshard_path_describer.cpp

```diff
@@ -72,6 +72,79 @@ static void FillTableStats(NKikimrSchemeOp::TPathDescription& pathDescription, c
     FillTableMetrics(pathDescription.MutableTabletMetrics(), stats);
 }
 
+static void FillColumns(
+    const TTableInfo& tableInfo,
+    google::protobuf::RepeatedPtrField<NKikimrSchemeOp::TColumnDescription>& out
+) {
+    bool familyNamesBuilt = false;
+    THashMap<ui32, TString> familyNames;
+
+    out.Reserve(tableInfo.Columns.size());
+    for (const auto& col : tableInfo.Columns) {
+        const auto& cinfo = col.second;
+        if (cinfo.IsDropped())
+            continue;
+
+        auto* colDescr = out.Add();
+        colDescr->SetName(cinfo.Name);
+        colDescr->SetType(NScheme::TypeName(cinfo.PType, cinfo.PTypeMod));
+        auto columnType = NScheme::ProtoColumnTypeFromTypeInfoMod(cinfo.PType, cinfo.PTypeMod);
+        colDescr->SetTypeId(columnType.TypeId);
+        if (columnType.TypeInfo) {
+            *colDescr->MutableTypeInfo() = *columnType.TypeInfo;
+        }
+        colDescr->SetId(cinfo.Id);
+        colDescr->SetNotNull(cinfo.NotNull);
+
+        if (cinfo.Family != 0) {
+            colDescr->SetFamily(cinfo.Family);
+
+            if (!familyNamesBuilt) {
+                for (const auto& family : tableInfo.PartitionConfig().GetColumnFamilies()) {
+                    if (family.HasName() && family.HasId()) {
+                        familyNames[family.GetId()] = family.GetName();
+                    }
+                }
+                familyNamesBuilt = true;
+            }
+
+            auto it = familyNames.find(cinfo.Family);
+            if (it != familyNames.end() && !it->second.empty()) {
+                colDescr->SetFamilyName(it->second);
```

Comment:
NIT:
FindPtr probably more convinient here

Reply:
I have copy-pasted this code. Don't want to dig into it